src/graph/expression_graph.h

```diff
@@ -441,38 +650,64 @@ class ExpressionGraph : public std::enable_shared_from_this<ExpressionGraph> {
     return it->second;
   }
 
+  /**
+   * Set default element type for the graph.
+   * The default value is used if some node type is not specified.
+   */
   void setDefaultElementType(Type defaultElementType) {
     ABORT_IF(!paramsByElementType_.empty() && defaultElementType != defaultElementType_, 
              "Parameter objects already exist, cannot change default type from {} to {}", 
              defaultElementType_, defaultElementType);
     defaultElementType_ = defaultElementType;
   }
 
+  /**
+   * Add a expression node to the graph.
+   * @param node a pointer to a expression node
+   */
   Expr add(Expr node);
 
+  /**
+   * Allocate memory for the forward pass of the given node.
+   * @param node a pointer to a expression node
+   */
   void allocateForward(Expr node) {
     if(tensors_)
       tensors_->allocateForward(node);
   }
 
+  /**
+   * Allocate memory for the backward pass of the given node.
+   * @param node a pointer to a expression node
+   */
   void allocateBackward(Expr node) {
     if(tensors_)
       tensors_->allocateBackward(node);
   }
 
+  /**
+   * Free the memory for a tensor object.
+   * @param tensor a reference to the tensor object
+   */
   void free(const Tensor& tensor) {
     if(tensors_)
       tensors_->free(tensor);
   }
 
-  // Returns the memory allocator of the graph workspace, allocates row unstructured memory (but 256-byte aligned)
+  /**
+   * Returns the memory allocator of the graph workspace,
+   * allocates raw unstructured memory (but 256-byte aligned).
```

Comment:
If we use automatic `@brief`, maybe these should be two sentences? So "Allocates raw unstructured..." should be the second one?

Reply:
No, doxygen will interpret the first line (**until the first dot**) of a Javadoc-style comment as the brief description. Those two line will be one sentence.

Reply:
I was suggesting that "Return the memory allocator of the graph workspace." is a good brief description and "Allocates raw unstructured memory (but 256-byte aligned)." can be in the full description.

Reply:
No problem. I will fix this.